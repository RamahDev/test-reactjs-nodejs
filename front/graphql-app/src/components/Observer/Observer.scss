.Observer {
    height: 100px;
}