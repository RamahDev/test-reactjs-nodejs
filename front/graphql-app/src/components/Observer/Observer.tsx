import React from "react";
import "./Observer.scss";

const Observer = React.forwardRef<any, any>((props, ref) => {
  return <div className="Observer" ref={ref}></div>;
});

export default Observer;
