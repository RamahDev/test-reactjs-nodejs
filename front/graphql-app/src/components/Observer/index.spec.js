import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Observer from './Observer';

const toggleExpand = jest.fn();
const defaultProps = {
  title: 'awesome title',
  content: {},
  onClick: toggleExpand,
};

test('<Observer /> component renders correctly', () => {
  const { container } = render(<Observer {...defaultProps} />);
  expect(container).toMatchSnapshot();
});
