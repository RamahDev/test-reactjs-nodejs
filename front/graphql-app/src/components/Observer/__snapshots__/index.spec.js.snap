// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Observer /> component renders correctly 1`] = `
<div>
  <div
    class="Observer"
  />
</div>
`;
